packages/relay-runtime/network/RelayObservable.js

```diff
@@ -477,6 +500,23 @@ function fromValue<T>(value: T): RelayObservable<T> {
   });
 }
 
+function fromAsyncIterator<T>(obj: AsyncIterator<T>): RelayObservable<T> {
+  return RelayObservable.create(sink => {
+    (async () => {
+      while (true) {
+        const {value, done} = await obj.next();
+        if (done) {break;}
+        sink.next(value);
+      }
+      sink.complete();
+    })().catch(error => {
+      sink.error(error, true /* isUncaughtThrownError */);
+    });
+
+    return () => obj.return?.();
```

Comment:
will this cause the loop above to break? or do we have to do that manually? i'm not familiar w details of generators

Reply:
Yeah that's right, if you return the generator the next while iteration will have a done true.